[박소현] Week3

ParkSohyunee opened the pull request:
## 요구사항

### 기본

- [x] [기본] 브라우저에 현재 보이는 화면의 영역(viewport) 너비를 기준으로 분기되는 반응형 디자인을 적용했나요?
PC: 1200px 이상
Tablet: 768px 이상 ~ 1199px 이하
Mobile: 375px 이상 ~ 767px 이하
375px 미만 사이즈의 디자인은 고려하지 않습니다
- [x] [랜딩 페이지] 페이스북, 카카오톡, 디스코드, 트위터 등 SNS에서 Linkbrary 랜딩 페이지(“/”) 공유 시 미리보기를 볼 수 있도록 랜딩 페이지 메타 태그를 설정했나요?
- [x] [랜딩 페이지] 미리보기에서 제목은 “Linkbrary”, 설명은 “세상의 모든 정보를 쉽게 저장하고 관리해 보세요”로 설정했나요?
- [x] [랜딩 페이지] 주소와 이미지는 자유롭게 설정했나요?
- [x] [랜딩 페이지] Tablet 사이즈에서 화면의 너비가 1199px 이하로 작아질 때 “Linkbrary” 로고와 “로그인” 버튼 사이의 간격은 변하지 않게 고정값으로 유지하되 좌우 여백이 줄어드나요?
- [x] [랜딩 페이지] Tablet 사이즈에서 최소 좌우 여백은 “Linkbrary” 로고의 왼쪽에 여백 32px, “로그인” 버튼 오른쪽 여백 32px을 유지하고“Linkbrary” 로고와 “로그인" 버튼의 간격이 가까워지나요?
- [x] [랜딩 페이지] PC, Tablet 사이즈의 이미지 크기는 고정값을 사용했나요?
- [x] [랜딩 페이지] Mobile 사이즈의 이미지는 좌우 여백 32px을 제외하고 이미지 영역이 꽉 차나요? (이때 가로가 커지는 비율에 맞춰 세로도 커져야 합니다.)
- [x] [랜딩 페이지] Mobile 사이즈 너비가 커지면, “Privacy Policy”, “FAQ”, “codeit-2023”이 있는 영역과 SNS 아이콘들이 있는 영역의 간격이 커지나요?
- [x] [로그인, 회원가입 페이지] Tablet 사이즈에서 디자인은 PC사이즈와 동일한가요?
- [x] [로그인, 회원가입 페이지] Mobile 사이즈에서 좌우 여백 32px 제외하고 내부 요소들이 너비를 모두 차지하나요?
- [x] [로그인, 회원가입 페이지] Mobile 사이즈에서 내부 요소들의 너비는 기기의 너비가 커지는 만큼 커지지만 400px을 넘지 않나요?

### 심화

- [x] 팀원의 위클리 미션 PR에 코드 리뷰를 남겼나요?
- [x] 랜딩 페이지 Mobile 사이즈에서 제품 소개 영역의 순서를 제목, 설명, 이미지 => 제목, 이미지, 설명 순서로 변경했나요?

## 주요 변경사항

- 반응형 UI 구현
- 랜딩페이지 main/content, footer 영역 grid 적용

## 스크린샷
> 모바일
<img width="209" alt="mobile" src="https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/c53a3147-fe1b-439b-811a-37f3adccb880">

> 태블릿
<img width="453" alt="tablet" src="https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/6029fe77-fbad-4392-8053-5bd458d87123">

> OG
<img width="284" alt="og" src="https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/bdcb30d0-69ee-4800-92af-e545a669af0a">

## 멘토에게

- 반응형 웹 디자인 적용 및 기타 코드 관련해서 확인 부탁드립니다.
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

ParkSohyunee committed:
style: header, hero, contents 반응형 UI 구현(태블릿, 모바일)

ParkSohyunee committed:
refactor: content 레이아웃 grid로 수정

ParkSohyunee committed:
style: main 반응형 UI 구현(태블릿, 모바일)

ParkSohyunee committed:
refactor: footer 레이아웃 grid 적용

ParkSohyunee committed:
style: footer 반응형 UI 구현(태블릿, 모바일)

ParkSohyunee committed:
style: 로그인, 회원가입페이지 반응형 UI 구현(모바일)

ParkSohyunee committed:
add: OG data

ParkSohyunee committed:
style: 주석 삭제

ParkSohyunee commented on Linkbrary/css/style.css:
```diff
@@ -167,5 +181,155 @@
+  .header__login__btn {
+    width: 80px;
+    padding: 10px 1rem;
+    font-size: 14px;
```
rem으로 변경할 경우 숫자가 애매한 것 같아서 px을 사용했습니다.
하지만 rem과 px을 같이 사용하니 가독성이 안좋아 보이네요.. 해당 부분 리팩토링 시에 수정반영하겠습니다!

ParkSohyunee replied:
멘토님 질문이 있습니다!!
오늘 멘토링 시간에 <br />태그 대신 <span>태그 사용 여쭤보신 부분이 
hero 영역 "세상의 모든 정보를 쉽게 저장하고 관리해 보세요" 이 부분 말씀하시는 것이 맞을까요?!

원래는 <br />태그를 사용하여 줄바꿈을 적용했는데 반응형을 적용할 때 "쉽게 저장하고 관리해 보세요" 문장이 
"쉽게 저장하고 <br /> 관리해 보세요" 문장으로 변경되어야 하는 부분때문에 <br />태그를 삭제하고 span태그로 영역을 
2개로 분리해서 반응형일때는 flex: column 값을 주는 것으로 구현했습니다. (다른 방법은 생각이 안났습니다 ㅠㅠ)

          <div class="hero__title--line">
            <span>쉽게 저장하고 </span><span>관리해 보세요</span>
          </div>

혹시 <br />태그를 삭제하지 않고 더 간단하게 해당 부분의 줄바꿈을 구현할 수 있는 방법이 있을까요?! 🥲

llucas93 replied:
네 소현님이 작성하신 방법도 좋은방법인데요
br로 사용한다면 max-width 값 과 word-break 같은 줄바꿈 속성 이용해서 구현해보면 어떨까 하는데요
아래는 제가 테스트로 작성해본 코드라 참고용도로만 확인해주시면 좋을것같습니다 👍 

```
<h1 class="hero__title">
  <span class="title__gradient">세상의 모든 정보를</span>
  <br />쉽게 저장하고 관리해 보세요
</h1>
```

``` 
.hero__title {
  white-space: no-wrap; <-- 제거
}

.title__gradient {
  // ...
  white-space: no-wrap; <-- 추가
}

/* Tablet */
@media only screen and (min-width: 768px) and (max-width: 1199px) {
  // ...
  .hero__title {
    margin: 0 auto;
    max-width: 472px;
    width: 100%;
    word-break: keep-all;
    overflow-wrap: break-word;
  }

  .hero__title > span {
    text-wrap: nowrap;
  }
}

/* Mobile */
@media only screen and (min-width: 375px) and (max-width: 767px) {
    .hero__title {
    // ...

    margin: 0 auto;
    max-width: 236px;
    width: 100%;
    word-break: keep-all;
    overflow-wrap: break-word;
  }

  .hero__title > span {
    text-wrap: nowrap;
  }
}
```

ParkSohyunee replied:
안녕하세요 멘토님!!! 상세히 확인 해 주셔서 감사합니다.😆
overflow-wrap 속성이 있는 줄 처음 알았는데 줄바꿈 시 해당 속성과 word-break 속성으로 개선할 수 있는 방법이 있었군요.🧐

전달 주신 참고 코드와 팀원분들의 의견 참고하여 리팩토링 적용해 보았습니다.
코드의 주요 수정사항은 다음과 같습니다. 확인 부탁드립니다.

- hero 이미지 사이즈 가로, 세로 100% 적용(페이지 너비에 따라 이미지가 꽉 차보이게 반영)
- hero title 영역 width 수정하여 줄바꿈 개선
- 반응형 font-size 및 일부 px 단위 rem으로 수정
- 불필요한 span태그 제거
- 주석삭제

seoyoung-min reviewed (commented):
소현님 파일명, 클래스명, 파일구성까지 너무 깔끔해서 코드가 정말 잘 읽혔습니다!
새로운 방식들이나, 몰랐던 속성값들(ex. grid에서 칸 수를 span으로 나타내기 등)을 알 수 있어서 많이 배웠습니다 ☺️

기타 의견을 남겼는데, 참고로 봐주시면 좋을 것 같습니다 :)

ParkSohyunee committed:
refactor: 메인페이지 hero, content 리팩토링
- hero 이미지 사이즈 100% 적용
- hero title 영역 width 수정하여 줄바꿈 개선
- 반응형 font-size 및 일부 px 단위 rem으로 수정
- 불필요한 span태그 제거
- 주석삭제
